packages/jitm/src/JITM.php

```diff
@@ -149,6 +153,116 @@ static function jitm_jetpack_woo_services_activate( $CTA ) {
 		);
 	}
 
+	/**
+	 * This is an entire admin notice dedicated to messaging and handling of the case where a user is trying to delete
+	 * the connection owner.
+	 */
+	function delete_user_update_connection_owner_notice() {
+		global $current_screen;
+
+		if ( 'users' !== $current_screen->base ) {
+			return;
+		}
+
+		if ( 'delete' !== $_REQUEST['action'] ) {
+			return;
+		}
+
+		// Get connection owner or bail
+		$connection_manager = new Manager();
+		$connection_owner   = $connection_manager->get_connection_owner();
+		if ( ! $connection_owner ) {
+			return;
+		}
+
+		// Bail if we're not trying to delete connection owner.
+		$user_ids_to_delete        = $_REQUEST['users'];
+		$deleting_connection_owner = in_array( $connection_owner->ID, (array) $user_ids_to_delete );
+		if ( ! $deleting_connection_owner ) {
+			return;
+		}
+
+		$connection_manager = new Manager();
+		$connected_users    = $connection_manager->get_connected_users( 'jetpack_connect' );
+
+		echo "<div class='notice notice-warning' id='jetpack-notice-switch-connection-owner'>";
+		_e( '<h2>Important notice about your Jetpack connection: </h2>', 'jetpack' );
+		printf(
+			__( "<p>Warning! You are about to delete the Jetpack <a href='%1\$1s' target='_blank'>connection owner</a> (%2\$2s) for this site, which may affect some of your features.</p>", 'jetpack' ),
+			'https://jetpack.com/support/primary-user/',
+			esc_html( $connection_owner->data->user_login )
+		);
+
+		if ( ! empty( $connected_users ) && count( $connected_users ) > 1 ) {
+			echo '<form id="jp-switch-connection-owner" action="" method="post">';
+			echo "<label for='owner'>" . __( 'You can choose to transfer connection ownership to one of these already-connected admins: ', 'jetpack' ) . '</label>';
+
+			$connected_user_ids = array_map(
+				function( $connected_user ) {
+						return $connected_user->ID;
+				},
+				$connected_users
+			);
+
+			wp_dropdown_users(
+				array(
+					'name'    => 'owner',
+					'include' => array_diff( $connected_user_ids, array( $connection_owner->ID ) ),
+					'show'    => 'display_name_with_login',
+				)
+			);
+
+			submit_button( __( 'Set new connection owner', 'jetpack' ), 'primary', 'jp-switch-connection-owner-submit' );
+
+			_e( '<p>As always, feel free to <a href="https://jetpack.com/contact-support" target="_blank">contact our support team</a> if you have any questions.</p>', 'jetpack' );
```

Comment:
Nit; this and other `target="_blank"` links would need `rel="noreferrer noopener"`

Reply:
No idea if this is possible easily but would be nice to have "external" icon after `_blank` links.

Reply:
Done for all, thanks!

Reply:
Yup 👍  Totally not a blocker, I did mean to mention that. :-) Lots of places in Jetpack seem to miss it already, too.